### ISSUE TITLE
Empty response when calling method  ArticleGet

### ISSUE THREAD

**Issue by Frankst2** [gtudan/OTRS-Client#7]
First off: Great work, the SOAP interface of OTRS isn't that easy to handle. This library certainly makes it easier.
Now to the problem: I wanted to use the TicketObject->ArticleGet method of OTRS 5.0.16 to get all articles attached to a ticket. However, the returned SOAPMessage is more or less empty. Is this a problem of this library or how SOAP was implemented by OTRS? From what I see with Wireshark OTRS does actually return something, but when I manually inspect the SOAPMessage at https://github.com/gtudan/OTRS-Client/blob/8296a077772b531bfa287d7f0120c4617555d608/src/main/java/de/tudan/otrsclient/OtrsConnector.java#L128 I see a null message body:
```
answerMsg.getSOAPBody().toString()
[soap:Body: null]
```

The XML returned looks like this:
```
<?xml version="1.0" encoding="UTF-8"?>
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/" xmlns:soapenc="http://schemas.xmlsoap.org/soap/encoding/" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" soap:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/">
   <soap:Body>
      <DispatchResponse xmlns="/Core">
         <s-gensym3>
            <UntilTime xsi:type="xsd:int">0</UntilTime>
            <Priority xsi:type="xsd:string">2 low</Priority>
            <SLA xsi:type="xsd:string" />
            <Cc xsi:type="xsd:string" />
            <TicketID xsi:type="xsd:int">86</TicketID>
            <CustomerUserID xsi:type="xsd:string">...</CustomerUserID>
            <ArticleTypeID xsi:type="xsd:int">8</ArticleTypeID>
            <EscalationResponseTime xsi:type="xsd:int">0</EscalationResponseTime>
            <QueueID xsi:type="xsd:int">6</QueueID>
            <Age xsi:type="xsd:int">430159</Age>
            <SLAID xsi:nil="true" />
            <IncomingTime xsi:type="xsd:int">1486044774</IncomingTime>
            <CreatedBy xsi:type="xsd:int">2</CreatedBy>
            <SenderType xsi:type="xsd:string">agent</SenderType>
            <State xsi:type="xsd:string">acknowledged</State>
            <EscalationSolutionTime xsi:type="xsd:int">0</EscalationSolutionTime>
            <FromRealname xsi:type="xsd:string">...</FromRealname>
            <TicketNumber xsi:type="xsd:long">2017020210000086</TicketNumber>
            <From xsi:type="xsd:string">...</From>
            <Subject xsi:type="xsd:string">...</Subject>
            <PriorityID xsi:type="xsd:int">2</PriorityID>
            <References xsi:type="xsd:string" />
            <StateID xsi:type="xsd:int">19</StateID>
            <RealTillTimeNotUsed xsi:type="xsd:int">0</RealTillTimeNotUsed>
            <ServiceID xsi:nil="true" />
            <InReplyTo xsi:type="xsd:string" />
            <Charset xsi:type="xsd:string">utf8</Charset>
            <CustomerID xsi:type="xsd:string">...</CustomerID>
            <ToRealname xsi:type="xsd:string">...</ToRealname>
            <EscalationTime xsi:type="xsd:int">0</EscalationTime>
            <MessageID xsi:type="xsd:string" />
            <SenderTypeID xsi:type="xsd:int">1</SenderTypeID>
            <MimeType xsi:type="xsd:string">text/plain</MimeType>
            <EscalationUpdateTime xsi:type="xsd:int">0</EscalationUpdateTime>
            <Service xsi:type="xsd:string" />
            <Owner xsi:type="xsd:string">...</Owner>
            <Created xsi:type="xsd:string">2017-02-02 15:12:54</Created>
            <Queue xsi:type="xsd:string">...</Queue>
            <StateType xsi:type="xsd:string">open</StateType>
            <CreateTimeUnix xsi:type="xsd:int">1486044774</CreateTimeUnix>
            <ResponsibleID xsi:type="xsd:int">1</ResponsibleID>
            <Title xsi:type="xsd:string">string</Title>
            <AgeTimeUnix xsi:type="xsd:int">430159</AgeTimeUnix>
            <ContentType xsi:type="xsd:string">text/plain; charset=utf8</ContentType>
            <To xsi:type="xsd:string">Product ordering</To>
            <Responsible xsi:type="xsd:string">...</Responsible>
            <OwnerID xsi:type="xsd:int">2</OwnerID>
            <Type xsi:type="xsd:string">Unclassified</Type>
            <Changed xsi:type="xsd:string">2017-02-02 15:12:54</Changed>
            <ArticleType xsi:type="xsd:string">webrequest</ArticleType>
            <ReplyTo xsi:type="xsd:string" />
            <Body xsi:type="xsd:string">...</Body>
            <ContentCharset xsi:type="xsd:string">utf8</ContentCharset>
            <Lock xsi:type="xsd:string">unlock</Lock>
            <ArticleID xsi:type="xsd:int">133</ArticleID>
            <TypeID xsi:type="xsd:int">1</TypeID>
            <LockID xsi:type="xsd:int">1</LockID>
         </s-gensym3>
      </DispatchResponse>
   </soap:Body>
</soap:Envelope>
```

For now I can probably work around this issue (I'll use the REST API).

**Comment by Frankst2** [comment]
BTW: I have the same problem when calling the LinkObject->LinkList method. I guess the SOAP code needs to be modified in some way? When I call System.out.println(msg.getSOAPPart().getEnvelope()); the result is "[soap:Envelope: null]"

**Comment by gtudan** [comment]
This sounds very much like a namespace issue - maybe related to the this PR? https://github.com/gtudan/OTRS-Client/pull/5

**Comment by Frankst2** [comment]
I'll try to fix this locally, PR #5 doesn't seem to be work in this case as I already tried it locally/pulled the latest code (also see my comment at https://github.com/gtudan/OTRS-Client/pull/5#issuecomment-279710325, I applied the patch to get a working ArticleSend method).

**Comment by Frankst2** [comment]
For reference: The DEBUG log info from OTRSClient with the request and the response for the LinkList call. I use OTRS ITSM 5.0.16.

Could the problem be related to this element, at least with LinkList issue (maybe the ArticleGet one is different)?
`<Target xsi:type="apachens:Map">`

Request:
```xml
<SOAP-ENV:Envelope  
    xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/"  
    xmlns:env="http://schemas.xmlsoap.org/soap/envelope/"  
    xmlns:soapenc="http://schemas.xmlsoap.org/soap/encoding/"  
    xmlns:xsd="http://www.w3.org/2001/XMLSchema"  
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"   SOAP-ENV:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/">  
    <SOAP-ENV:Header/>  
    <SOAP-ENV:Body>  
        <Dispatch  
            xmlns="/Core">  
            <Username xsi:type="xsd:string">...</Username>  
            <Password xsi:type="xsd:string">...</Password>  
            <Object xsi:type="xsd:string">LinkObject</Object>  
            <Method xsi:type="xsd:string">LinkList</Method>  
            <Param0_Name xsi:type="xsd:string">State</Param0_Name>  
            <Param0_Value xsi:type="xsd:string">Valid</Param0_Value>  
            <Param1_Name xsi:type="xsd:string">UserID</Param1_Name>  
            <Param1_Value xsi:type="xsd:integer">1</Param1_Value>  
            <Param2_Name xsi:type="xsd:string">Object</Param2_Name>  
            <Param2_Value xsi:type="xsd:string">ITSMConfigItem</Param2_Value>  
            <Param3_Name xsi:type="xsd:string">Key</Param3_Name>  
            <Param3_Value xsi:type="xsd:string">11</Param3_Value>  
        </Dispatch>  
    </SOAP-ENV:Body>  
</SOAP-ENV:Envelope>
```

Response:
```xml
<?xml version="1.0" encoding="UTF-8"?>  
<soap:Envelope soap:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"  
    xmlns:apachens="http://xml.apache.org/xml-soap"  
    xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/"  
    xmlns:soapenc="http://schemas.xmlsoap.org/soap/encoding/"  
    xmlns:xsd="http://www.w3.org/2001/XMLSchema"  
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">  
    <soap:Body>  
        <DispatchResponse  
            xmlns="/Core">  
            <s-gensym9>  
                <ITSMConfigItem>  
                    <DependsOn>  
                        <Target xsi:type="apachens:Map">  
                            <item>  
                                <key xsi:type="xsd:int">12</key>  
                                <value xsi:type="xsd:int">1</value>  
                            </item>  
                        </Target>  
                    </DependsOn>  
                </ITSMConfigItem>  
            </s-gensym9>  
        </DispatchResponse>  
    </soap:Body>  
</soap:Envelope>
```

**Comment by Frankst2** [comment]
Sorry about the last few comments, it looks like I used OTRSClient in the somewhat wrong way (not 100% comfortable with SOAP yet). I think I need to walk/parse the response myself as the ITSMConfigItem response does not match what OTRSClient expects? After all I need to parse the "DependsOn" information etc.

I think we can close this issue for now, I realized what I wrote in my first comment was wrong (answerMsg.getSOAPBody().toString() returned null as this element has no value).
